src/components/assignment-form/input-output.jsx

```diff
@@ -88,6 +103,35 @@ class InputOutput extends Component {
       </div>
     );
   }
+
+  renderReadOnly() {
+    return (
+      <div >
+        <div className={prefixer('field-container')}>
+          <input
+            readOnly
+            aria-label="testisyöte"
+            className={prefixer('input-field')}
+            type="text"
+            value={this.props.io.input}
+          />
+          <div className={prefixer('test-field-arrow')}>&rarr;
+          </div>
+          <input
+            readOnly
+            aria-label="testituloste"
+            className={prefixer('input-field')}
+            type="text"
+            value={this.props.io.output}
+          />
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    return this.props.readOnly ? this.renderReadOnly() : this.renderEditable();
```

Comment:
If editable and read-only variants are that different does it make sense to extract these into their own components?